products/monitoring/api.yaml

```diff
@@ -886,6 +886,7 @@ objects:
     - !ruby/object:Api::Type::String
       name: period
       default_value: "300s"
+      input: true
```

Comment:
@chrisst I was about to submit exactly the same change as a PR, but it did not work for me locally. The problem with `input: true` was that `period` was only included in the JSON when the resource was created, and omitted when the resource was patched (for example, when I only changed `timeout`). And the resource API rejects any patch requests if `period` is missing, in the same way as it rejects them when the `period` is different from the original. So it seems we always need to send `period` down to the API, but create a new resource when the period changes. So I ended up leaving the [comment](https://github.com/terraform-providers/terraform-provider-google/issues/2703#issuecomment-450577202) hoping you'd figure out how to make it work properly.
The way to reproduce: create the resource, change only the `timeout` value, `terraform apply`.

Reply:
Ah great catch, thanks. That will teach me to run the integration tests before pushing :P

Reply:
It may as well be a quirk of Google Cloud API. I noticed that when you edit the policy in the Cloud Console, it sends requests directly to StackDriver API. So I guess Google Cloud API probably delegates to StackDriver API behind the scenes, and something may end up being lost in translation.

Reply:
I had to take a closer look at the docs but it looks like they have overloaded the PATCH end point so that it can perform both PUT and PATCH behavior based on the presence of the update mask. From the Update Mask field description:
"If this field is empty, then the current configuration is completely replaced with the new configuration."